test/server/cards/01_SOR/events/IHadNoChoice.spec.ts

```diff
@@ -0,0 +1,92 @@
+describe('I Had No Choice', function() {
+    integration(function(contextRef) {
+        describe('I Had No Choice\'s ability', function() {
+            describe('when there are two or more non-leader units in play', function() {
+                beforeEach(function () {
+                    contextRef.setupTest({
+                        phase: 'action',
+                        player1: {
+                            hand: ['i-had-no-choice'],
+                            deck: ['foundling', 'pyke-sentinel', 'atst', 'cartel-spacer', 'battlefield-marine'],
```

Comment:
In the future it's fine to just use duplicates of 'atst' as a filler card if you need to pad stuff out, just to save typing time